src/streamdatasetjson/datasetjson.py

```diff
@@ -1,22 +1,46 @@
+from typing import List
+import ijson
 import utils
 
 from dataset import Dataset
 from errors import DatasetNotFoundError
 
 
 class DatasetJSON:
+    """
+    Dataset JSON instance. 
+    """
     def __enter__(self):
         self._file = open(self._path, "r")
+        self._dataset_prefixes = self._generate_dataset_prefixes()
         return self
 
     def __exit__(self, type, value, traceback):
-        self._file.close()
+        self.close()
 
     def __init__(self, path):
         self._path = path
 
-    def get_dataset(self, target_name):
-        try:
-            return Dataset(self._file, target_name)
-        except StopIteration:
-            raise DatasetNotFoundError(target_name)
\ No newline at end of file
+    def get_dataset(self, target_name: str) -> Dataset:
+        if target_name not in self._dataset_prefixes.keys():
+            raise DatasetNotFoundError(target_name)
+
+        return Dataset(self._file, target_name, self._dataset_prefixes[target_name])
+    
+    @property
+    def available_datasets(self) -> List[str]:
+        return self._dataset_prefixes.keys()
+
+    def close(self):
```

Comment:
As I understand it is possible to use this class only with Context Management, do we need to expose this method? If we do not need to expose it, should the code be moved to to __exit__?

Reply:
In my opinion it is necessary. The possibility of using this class with context management is useful for users who are using it in a script. But in some cases users might prefer the default handling. 

For example, in our server we could caché 1 instance of this class for each file that was opened, by doing this we don't instance objects of this class every time a request is received and save some time (remember that the constructor is slow). So by having the option to not use the context manager, in our case, is more convenient.